lib/ui/common/roundedButton.dart

```diff
@@ -2,125 +2,156 @@ import 'package:flutter/material.dart';
 
 class RoundedButtonViewModel {
   String title;
-  IconData icon;
+  String icon;
   Function onTap;
-  BuildContext context;
 
-  RoundedButtonViewModel(this.context, {this.title, this.icon, this.onTap});
+  RoundedButtonViewModel({this.title, this.icon, this.onTap});
 }
 
-RoundedButtonViewModel buildButtonViewModel(BuildContext context, {String title, IconData icon, Function onTap}) {
-  return RoundedButtonViewModel(context, title: title, icon : icon, onTap: () => onTap(context));
-}
-
-abstract class RoundedButtonStyle {
-
+class RoundedButtonStyle {
   final Color iconButtonColor;
   final Color backgroundColor;
-
   final BoxShape buttonShape;
   final BorderRadius borderRadius;
-
   final EdgeInsets edgePadding;
   final TextStyle buttonTextStyle;
-
-  final double size;
+  final double height;
+  final double width;
   final double iconEdgePadding;
   final double buttonIconSize;
 
-  RoundedButtonStyle(this.size,
-      this.iconButtonColor, this.backgroundColor, this.iconEdgePadding,
-      {this.buttonShape, this.borderRadius, this.edgePadding,
-        this.buttonTextStyle, this.buttonIconSize});
-}
-
-class _DefaultStyle implements RoundedButtonStyle {
-
-  final Color iconButtonColor =  const Color(0xFFFFFFFF);
-  final Color backgroundColor =  const Color(0xff24d900);
-
-  final BoxShape buttonShape = BoxShape.rectangle;
-  final BorderRadius borderRadius = const BorderRadius.all(Radius.circular(20));
-
-  final EdgeInsets edgePadding = const EdgeInsets.only(left: 32, top: 31, right: 32, bottom: 32);
-  final TextStyle buttonTextStyle = const TextStyle(fontSize: 17, fontWeight: FontWeight.bold, color: Color(0xffffffff));
+  RoundedButtonStyle({
+    this.height,
+    this.width,
+    this.iconButtonColor,
+    this.backgroundColor,
+    this.iconEdgePadding,
+    this.buttonShape,
+    this.borderRadius,
+    this.edgePadding,
```

Comment:
I think that we shouldn't have the padding inside the widget, the padding should be set on the parent widget that contain this parent. Because If I want to to use this widget I'll have this default padding and maybe I want to have less padding.